### PR TITLE
Configure Radium package to use externals as set at build time

### LINKED ISSUE #618: Radium external not found when compiled separately
Since the merge of the PR #554, one might compile and install external dependencies independently of RAdium Bundl.
When doing this, client application/lib/plugins ... do not have access to the Radium external dependencies without redefining their location (using, e.g. Eigen3_DIR ...)

Setting these locations is error-prone and might be confusing (they are several, see below) for Radium user.

The following variable might be set automatically by the Radium package to the values used when compiling Radium :  
```
Eigen3_DIR    
OpenMesh_DIR  
cpplocate_DIR 
glm_DIR       
glbinding_DIR 
globjects_DIR 
stb_DIR       
assimp_DIR    
tinyply_DIR   
```

### PR DESCRIPTION
**Pull request by MathiasPaulin**
Closes #618

This PR register, in the Radium package, the following variables (the location of the installed Radium dependencies) at the value they have when building Radium.
These values could be into the Radium Bundle, a system-wide location or a specific installation location.

```
Eigen3_DIR    
OpenMesh_DIR  
cpplocate_DIR 
glm_DIR       
glbinding_DIR 
globjects_DIR 
stb_DIR       
assimp_DIR    
tinyply_DIR   
```

This allows client app/lib/plugin to only search for the package Radium (using find_package) to have access to any needed Radium external dependency.

### CODE REVIEW

**Comment by nmellado** [comment]
Hum, cannot find Eigen when compiling PluginUpstreamTarget
https://travis-ci.com/github/STORM-IRIT/Radium-Engine/jobs/442515296#L8788

File: cmake/Config.cmake.in [hunk 1]
```diff
@@ -30,9 +30,22 @@ if(@OPENMP_FOUND@) # Indicates if OpenMP has been used when compiling the Radium
     find_dependency(OpenMP REQUIRED)
 endif()
 # Theses paths reflect the paths founds in RadiumEngine/src/Core/external/package
```
Comment (dlyr) [inline cmake/Config.cmake.in:32]:
```suggestion
# Theses paths reflect the paths founds in RadiumEngine/external/Core/package
```

File: cmake/Config.cmake.in [hunk 2]
```diff
@@ -65,9 +78,21 @@ endforeach()
 
 if(Engine_FOUND)
     # Theses paths reflect the paths founds in RadiumEngine/src/Engine/external/package
```
Comment (dlyr) [inline cmake/Config.cmake.in:80]:
```suggestion
    # Theses paths reflect the paths founds in RadiumEngine/external/Engine/package
```

File: cmake/Config.cmake.in [hunk 3]
```diff
@@ -85,7 +110,11 @@ if(IO_FOUND)
 
     # Theses paths reflect the paths founds in RadiumEngine/src/IO/external/package
```
Comment (dlyr) [inline cmake/Config.cmake.in:111]:
```suggestion
    # Theses paths reflect the paths founds in RadiumEngine/external/IO/package
```

File: cmake/Config.cmake.in [hunk 4]
```diff
@@ -30,9 +30,22 @@ if(@OPENMP_FOUND@) # Indicates if OpenMP has been used when compiling the Radium
     find_dependency(OpenMP REQUIRED)
 endif()
 # Theses paths reflect the paths founds in RadiumEngine/src/Core/external/package
-set(Eigen3_DIR    "${CMAKE_CURRENT_LIST_DIR}/../../../@Eigen3_sub_DIR@")
-set(OpenMesh_DIR  "${CMAKE_CURRENT_LIST_DIR}/../../../@OpenMesh_sub_DIR@")
-set(cpplocate_DIR "${CMAKE_CURRENT_LIST_DIR}/../../../@cpplocate_sub_DIR@")
+if ("@Eigen3_DIR@" STREQUAL "" )
```
Comment (dlyr) [inline cmake/Config.cmake.in:33]: Is there any way to automate this instead of having the information in two places ?

**Review by dlyr — CHANGES_REQUESTED** [review]
Please at least update comments,
and at most find a way to automate ;)

File: cmake/Config.cmake.in (hunk 4 above)
Comment (nmellado) [reply cmake/Config.cmake.in:33]: I think we already automated this a lot... the problem is that we need to forward information from config; build and install times, for both Radium and the externals.

**Comment by MathiasPaulin** [comment]
> Please at least update comments,
comments were updated according to your suggestion
> and at most find a way to automate ;)
It is automated as musch as I can for now.

What is the change you request ? If there is no more change to do, please, modify your review. This PR is important for Radium based app/lib and plugin dev.

**Review by dlyr — APPROVED** [review]

**Pull request merged**